### PR TITLE
 Implement Search Text Input in UIKit 

### PR DESCRIPTION
**Pull request by Rhotimee**
#### Summary
Search Text Input component implementation.

### CODE REVIEW

File: packages/components/inputs/search-text-input/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# @commercetools-uikit/view-switcher
+
+## 15.11.0
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/search-text-input/CHANGELOG.md:3]:
```suggestion
## 15.11.2
```

File: packages/components/inputs/search-text-input/src/search-text-input.spec.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    const { container } = render(
+      <SearchTextInput {...baseProps} data-foo="bar" />
+    );
+    expect(container.querySelector('input')).toHaveAttribute('data-foo', 'bar');
+  });
+
+  it('should render a number input', () => {
+    const { container } = render(<SearchTextInput {...baseProps} />);
+    expect(container.querySelector('input')).toHaveAttribute('type', 'text');
+  });
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/search-text-input/src/search-text-input.spec.js:37]: Not sure if the title of the test is accurate.
It seems we're just trying to render the component with an empty value and checking there's a `input` element rendered.

**Comment by CarlosCortizasCT** [comment]
Good job 👍 

I just wanted to leave here some things that might be adjusted after reviewing the component in the preview link.

Default theme:
* There's not hover effect (border should change color).
* Background color changes when disabled when it should not.
* The lens icon changes its color when the component is disabled/read-only. I think this should not happen.

New theme:
* For consistency, I think the lens icon should have a 12px right margin instead of 16px.
* I'm not sure if it's expected for the lens icon to be green when hovered.
* Background color changes when disabled when it should not.
* Background color changes when read only when it should not.
* For consistency, I think horizontal spacing between close and lens icons should be 8px

Maybe @FilPob can confirm some of this points.

**Comment by FilPob** [comment]
thanks for this!

the points Carlos mentioned are all correct.  
Additionally:

1. yes the search icon should be green on hover (is correct already)
2. I would imagine that the "clear" icon could be optional (same as we have in the select inputs with the `clearable` prop)
3. The "clear" icon should not be shown when disabled or read only

Otherwise everything looks good

File: packages/components/inputs/search-text-input/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# @commercetools-uikit/view-switcher
```
[the remark this answers is not in the document]
Comment (Rhotimee) [reply packages/components/inputs/search-text-input/CHANGELOG.md]: Updated [691dbf1](https://github.com/commercetools/ui-kit/pull/2411/commits/691dbf1fd79b3f449f279864fdc9fa5de2de9afc)

File: packages/components/inputs/search-text-input/CHANGELOG.md (hunk 1 above)
Comment (Rhotimee) [reply packages/components/inputs/search-text-input/CHANGELOG.md]: Updated [691dbf1](https://github.com/commercetools/ui-kit/pull/2411/commits/691dbf1fd79b3f449f279864fdc9fa5de2de9afc)

File: .changeset/weak-students-shop.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@commercetools-uikit/search-text-input': minor
+'@commercetools-uikit/inputs': minor
+---
+
+Add SearchTextInput component to ui-kit
```
[the remark this answers is not in the document]
Comment (Rhotimee) [reply .changeset/weak-students-shop.md]: Updated [5c9839c](https://github.com/commercetools/ui-kit/pull/2411/commits/5c9839c2cee580b8cb7b156d8cdeb98b07b1d30b)

File: packages/components/inputs/search-text-input/src/search-text-input.spec.js (hunk 2 above)
Comment (Rhotimee) [reply packages/components/inputs/search-text-input/src/search-text-input.spec.js]: Updated [f6ff5c6](https://github.com/commercetools/ui-kit/pull/2411/commits/f6ff5c6ad2b798eac4bc285afba163f835a36f8a)

**Comment by FilPob** [comment]
One little thing: The search icon should not change color when in read only state

**Comment by FilPob** [comment]
Great! looks good now

**Review by FilPob — APPROVED** [review]

File: packages/components/inputs/search-text-input/LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2019 commercetools GmbH
```
Comment (tdeekens) [inline packages/components/inputs/search-text-input/LICENSE:3]: You can remove the year

Comment (Rhotimee) [reply packages/components/inputs/search-text-input/LICENSE]: Updated [975e64c](https://github.com/commercetools/ui-kit/pull/2411/commits/975e64c4e6fe3f4fc12ded6f2dcf9c22b6ee339b)

File: packages/components/inputs/search-text-input/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+| `autoComplete`         | `string`                                                                                              |          |           | Used as HTML autocomplete property                                                                                        |
+| `aria-invalid`         | `boolean`                                                                                             |          |           | Indicate if the value entered in the input is invalid.                                                                    |
+| `aria-errormessage`    | `string`                                                                                              |          |           | HTML ID of an element containing an error message related to the input.                                                   |
+| `name`                 | `string`                                                                                              |          |           | Used as HTML name of the input component. property.                                                                       |
```
[the remark this answers is not in the document]
Comment (Rhotimee) [reply packages/components/inputs/search-text-input/README.md]: Updated [975e64c](https://github.com/commercetools/ui-kit/pull/2411/commits/975e64c4e6fe3f4fc12ded6f2dcf9c22b6ee339b)

**Review by tdeekens — APPROVED** [review]

**Pull request merged**